## Bayesian Analysis of Disease A

### Questions asked in the problems are: 

#### a) Finding the Likelihood Function
The likelihood $p(D|\mathbf{w})$ is the probability of given condition $\mathbf{w}$ and have event $D$ happen.
* $P(+|A) = 0.98$ (given the case that a person actually has disease A, how likely would the person tested positive?)
* $P(+|not A) = 0.03$ (given the case that a person does NOT have disease A, how likely would the person tested positive?)

#### b) Finding the Bayesian Prior
$$\text{posterior} \propto \text{likelihood} \times \text{prior}$$
$$p(\mathbf{w}|D) = \frac{p(D|\mathbf{w})p(\mathbf{w})}{p(D)}$$
The prior $p(\mathbf{w})$ in this case is $p(A)$ or $p(notA)$.
* $p(A) = 0.001$ (Probability of having disease A)
* $p(not A) = 0.999$ (Probability of not having disease A)

#### c) The Normalization Term
In the textbook it says, "... p(D) the normalization constant, which ensures that the posterior distribution on the left-hand side is a valid probability density and integrates to one." In this case, $p(D)$ is $p(+)$". 
<br>
Hence, We can intuitively say the below without doing the integral: 
$$p(+) = p(+|A)p(A) + p(+|not A)p(not A)$$
(which, I wrote down in my paper notes)
$$p(A|+) = \frac{p(+|A)p(A)}{p(+)}$$
$$p(not A|+) = \frac{p(+|not A)p(not A)}{p(+)}$$
<br>
Since $p(A|+) + p(not A|+) = 1$ (all positive cases), we rearrange to get the equation for $p(+)$. 

In [16]:
# Constants
prior_A = 0.001
prior_not_A = 1 - prior_A

likelihood_pos_A = 0.98
likelihood_pos_not_A = 0.03

# Calculate p(+)
prob_positive = (likelihood_pos_A * prior_A) + (likelihood_pos_not_A * prior_not_A)

# Calculate Posterior Probability P(A|+)
posterior_A = (likelihood_pos_A * prior_A) / prob_positive

print(f"Normalization term p(+): {prob_positive:.5f}")
print(f"Posterior p(A|+): {posterior_A:.5f}")

Normalization term p(+): 0.03095
Posterior p(A|+): 0.03166


### Conclusion
The posterior probability is about 3.17%, meaning that the chance of having disease A with a positive test results is about **3.17%**. This number is close to the false positive test results ($P(+|not A) = 0.03 ≈ 3 \%$). My interpretation of the results comparison is that, tf someone tested positive, they have about the same probability of having disease A or getting a false result. 